src/core/tests/type_prop/group_normalization.cpp

```diff
@@ -0,0 +1,193 @@
+    const auto scale = std::make_shared<opset12::Parameter>(element::f32, PartialShape{-1});
+    const auto bias = std::make_shared<opset12::Parameter>(element::f32, PartialShape{4});
+
+    const auto gn = std::make_shared<opset12::GroupNormalization>(data, scale, bias, 2, 0.00001f);
+    EXPECT_EQ(gn->get_element_type(), element::f32);
+    EXPECT_EQ(gn->get_output_partial_shape(0), (PartialShape{1, 4, 6, 6}));
+}
+
+TEST(type_prop, group_normalization_dynamic_bias) {
+    const auto data = std::make_shared<opset12::Parameter>(element::f32, PartialShape{1, 4, 6, 6});
+    const auto scale = std::make_shared<opset12::Parameter>(element::f32, PartialShape{4});
+    const auto bias = std::make_shared<opset12::Parameter>(element::f32, PartialShape{-1});
+
+    const auto gn = std::make_shared<opset12::GroupNormalization>(data, scale, bias, 2, 0.00001f);
+    EXPECT_EQ(gn->get_element_type(), element::f32);
+    EXPECT_EQ(gn->get_output_partial_shape(0), (PartialShape{1, 4, 6, 6}));
+}
+
+TEST(type_prop, group_normalization_dynamic_rank) {
+    const auto data = std::make_shared<opset12::Parameter>(element::f16, PartialShape::dynamic());
+    const auto scale = std::make_shared<opset12::Parameter>(element::f16, PartialShape{6});
+    const auto bias = std::make_shared<opset12::Parameter>(element::f16, PartialShape{6});
+
+    const auto gn = std::make_shared<opset12::GroupNormalization>(data, scale, bias, 3, 0.00001f);
+    EXPECT_EQ(gn->get_element_type(), element::f16);
+    EXPECT_EQ(gn->get_output_partial_shape(0), (PartialShape::dynamic()));
+}
+
+TEST(type_prop, group_normalization_dynamic_everything) {
+    const auto data = std::make_shared<opset12::Parameter>(element::f16, PartialShape{3, -1, 10, 10});
+    const auto scale = std::make_shared<opset12::Parameter>(element::f16, PartialShape{-1});
+    const auto bias = std::make_shared<opset12::Parameter>(element::f16, PartialShape{-1});
+
+    const auto gn = std::make_shared<opset12::GroupNormalization>(data, scale, bias, 7, 0.00001f);
+    EXPECT_EQ(gn->get_element_type(), element::f16);
+    EXPECT_EQ(gn->get_output_partial_shape(0), (PartialShape{3, -1, 10, 10}));
+}
+
+TEST(type_prop, group_normalization_dynamic_ranks) {
+    const auto data = std::make_shared<opset12::Parameter>(element::f16, PartialShape::dynamic());
+    const auto scale = std::make_shared<opset12::Parameter>(element::f16, PartialShape::dynamic());
+    const auto bias = std::make_shared<opset12::Parameter>(element::f16, PartialShape::dynamic());
+
+    const auto gn = std::make_shared<opset12::GroupNormalization>(data, scale, bias, 12, 0.00001f);
+    EXPECT_EQ(gn->get_element_type(), element::f16);
+    EXPECT_EQ(gn->get_output_partial_shape(0), (PartialShape::dynamic()));
+}
+
+TEST(type_prop, group_normalization_dynamic_intervals) {
+    const auto data = std::make_shared<opset12::Parameter>(element::f32, PartialShape{2, Dimension{10, 20}, 6, 6});
```

Comment:
Maybe add here labels to all shapes on input and check that only labels from  data inputs are propagated.